contracts/swapHandlers/SwapHandlerCombinedBase.sol

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+pragma solidity ^0.8.0;
+
+import "./SwapHandlerBase.sol";
+import "../vendor/ISwapRouterV3.sol";
+import "../vendor/ISwapRouterV2.sol";
+
+/// @notice Base contract for swap handlers which execute a secondary swap on Uniswap V2 or V3 for exact output
+abstract contract SwapHandlerCombinedBase is SwapHandlerBase {
+    address immutable public uniSwapRouterV2;
+    address immutable public uniSwapRouterV3;
+
+    constructor(address uniSwapRouterV2_, address uniSwapRouterV3_) {
+        uniSwapRouterV2 = uniSwapRouterV2_;
+        uniSwapRouterV3 = uniSwapRouterV3_;
+    }
+
+    function executeSwap(SwapParams memory params) external override {
+        if (params.mode == 0) {
+            swapPrimary(params);
+        } else {
+            // For exact output expect a payload for the primary swap provider and a path to swap the remainder on Uni2 or Uni3
+            bytes memory path;
+            (params.payload, path) = abi.decode(params.payload, (bytes, bytes));
+
+            uint primaryAmountOut = swapPrimary(params);
+
+            if (primaryAmountOut < params.amountOut) {
+                require(path.length > 0, "SwapHandlerPayloadBase: secondary path not provided");
```

Comment:
I would not require user to provide the `path` for additional swap. 
In the future we might have markets on Euler that do not require an asset having Uniswap pair. Then, if there's no pair on Uni v2 or v3 and the swap is made using i.e. 1inch, the `swapAndRepay` would never succeed even if the amount out meets the condition imposed by `exactOutTolerance`. Hence, I'd move the `path.length > 0` check to the `if` statement above and let the `SwapHub` decide if the swapped amount is sufficient or not

Reply:
that would be streaching the semantics of swapAndRepay a bit as it's meant to calculate the swap amount within the tx. In case where there's no secondary swap, imo its better to do swap & burn, even if it costs a bit more gas, otherwise the UI would need to be very careful with setting target debt and exact out tolerance and it still would be brittle due to slippage.

If we have tokens with no uni pairs, the nice thing about swap hub is that we can just add another handler.